org.adempiere.base/src/org/compiere/model/MColumn.java

```diff
@@ -561,6 +561,11 @@ && getAD_Element_ID() != 0)
 				return false;
 			}
 		}
+		
+		if (get_ValueAsInt(COLUMNNAME_AD_Reference_ID) == 20 && (!get_ValueAsBoolean(COLUMNNAME_IsMandatory) || get_Value(COLUMNNAME_DefaultValue) == null)) {
```

Comment:
should be `if (getAD_Reference_ID()==DisplayType.YesNo) && (!isMandatory() || Util.isEmpty(getDefaultValue(),true))` instead.